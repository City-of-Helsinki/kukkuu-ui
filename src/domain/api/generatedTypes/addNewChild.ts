/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AddChildMutationInput } from './globalTypes';

// ====================================================
// GraphQL mutation operation: addNewChild
// ====================================================

export interface addNewChild_addChild_child_project {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  year: number;
}

export interface addNewChild_addChild_child {
  /**
   * The ID of the object.
   */
  id: string;
  firstName: string;
  lastName: string;
  birthdate: any;
  postalCode: string;
  project: addNewChild_addChild_child_project;
}

export interface addNewChild_addChild {
  child: addNewChild_addChild_child | null;
}

export interface addNewChild {
  /**
   * This mutation cannot be used before one has started using the service with "SubmitChildrenAndGuardianMutation".
   */
  addChild: addNewChild_addChild | null;
}

export interface addNewChildVariables {
  input: AddChildMutationInput;
}
