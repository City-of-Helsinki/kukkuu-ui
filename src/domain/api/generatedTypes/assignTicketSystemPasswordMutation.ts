/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AssignTicketSystemPasswordMutationInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: assignTicketSystemPasswordMutation
// ====================================================

export interface assignTicketSystemPasswordMutation_assignTicketSystemPassword {
  /**
   * The assigned ticket system password
   */
  password: string | null;
}

export interface assignTicketSystemPasswordMutation {
  assignTicketSystemPassword: assignTicketSystemPasswordMutation_assignTicketSystemPassword | null;
}

export interface assignTicketSystemPasswordMutationVariables {
  input: AssignTicketSystemPasswordMutationInput;
}
