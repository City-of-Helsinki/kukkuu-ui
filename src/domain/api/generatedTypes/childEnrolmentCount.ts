/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: ChildEnrolmentCount
// ====================================================

export interface ChildEnrolmentCount_child_project {
  /**
   * The ID of the object.
   */
  id: string;
  /**
   * How many times a single user can participate events per year. Changing this will not affect any existing enrolments.
   */
  enrolmentLimit: number;
}

export interface ChildEnrolmentCount_child {
  /**
   * The ID of the object.
   */
  id: string;
  /**
   * How many past enrolments child has this year.
   */
  pastEnrolmentCount: number | null;
  project: ChildEnrolmentCount_child_project;
}

export interface ChildEnrolmentCount {
  child: ChildEnrolmentCount_child | null;
}

export interface ChildEnrolmentCountVariables {
  childId: string;
}
