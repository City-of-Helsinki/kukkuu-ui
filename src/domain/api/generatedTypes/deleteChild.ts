/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { DeleteChildMutationInput } from './globalTypes';

// ====================================================
// GraphQL mutation operation: deleteChild
// ====================================================

export interface deleteChild_deleteChild {
  clientMutationId: string | null;
}

export interface deleteChild {
  deleteChild: deleteChild_deleteChild | null;
}

export interface deleteChildVariables {
  input: DeleteChildMutationInput;
}
