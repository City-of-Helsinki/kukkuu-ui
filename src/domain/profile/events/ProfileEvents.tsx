import React, { FunctionComponent } from 'react';

import ProfileEventsList from './ProfileEventsList';
import ProfileNoEvent from './ProfileNoEvent';
import { childByIdQuery_child as ChildByIdResponse } from '../../api/generatedTypes/childByIdQuery';

interface ProfileEventsProps {
  child: ChildByIdResponse;
}

const ProfileEvents: FunctionComponent<ProfileEventsProps> = ({ child }) => {
  const hasEvents = (child: ChildByIdResponse) => {
    return child.availableEvents || child.enrolments || child.pastEvents;
  };

  return hasEvents(child) ? (
    // TODO: normalize? null?
    <ProfileEventsList
      availableEvents={child.availableEvents || undefined}
      enrolments={child.enrolments}
      pastEvents={child.pastEvents || undefined}
    />
  ) : (
    <ProfileNoEvent />
  );
};

export default ProfileEvents;
