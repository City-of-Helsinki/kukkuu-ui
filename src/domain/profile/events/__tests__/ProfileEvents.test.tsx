import ProfileEvents from '../ProfileEvents';
import { EventParticipantsPerInvite } from '../../../api/generatedTypes/graphql';
import { render } from '../../../../common/test/testingLibraryUtils';
import {
  ChildByIdResponse,
  UpcomingEventsAndEventGroups,
  PastEvents,
  InternalAndTicketSystemEnrolments,
} from '../../../child/types/ChildByIdQueryTypes';

vi.mock('react-qrcode-logo', async (importOriginal: any) => {
  const mod = await importOriginal();
  return {
    ...mod,
    QRCode: () => <div />,
  };
});

const childData: ChildByIdResponse = {
  id: '',
  name: '',
  birthyear: 0,
  postalCode: '',
  project: {
    id: '',
    year: 0,
    name: 'Test project',
  },
  relationships: {
    edges: [],
  },
  upcomingEventsAndEventGroups: { edges: [] },
  activeInternalAndTicketSystemEnrolments: { edges: [] },
  pastEvents: { edges: [] },
};

const childNoEvents = {
  ...childData,
  upcomingEvents: null,
  enrolments: {
    edges: [],
  },
  pastEvents: null,
};

const upcomingEventsAndEventGroups: UpcomingEventsAndEventGroups = {
  edges: [
    {
      node: {
        id: 'RXZlbnROb2RlOjE=',
        name: 'pentti',
        shortDescription: 'eventti',
        participantsPerInvite: EventParticipantsPerInvite.ChildAndGuardian,
        image:
          'http://localhost:8081/media/2020-02-15-184035_1920x1080_scrot.png',
        imageAltText: 'uooo',
        canChildEnroll: true,
        __typename: 'EventNode',
      },
    },
  ],
};

const enrolments: InternalAndTicketSystemEnrolments = {
  edges: [
    {
      node: {
        __typename: 'EnrolmentNode',
        id: 'foo',
        referenceId: 'bar',
        occurrence: {
          id: '',
          time: '2020-02-24T07:07:18+00:00', // 09.07
          venue: {
            id: '',
            address: '',
            name: '',
          },
          event: {
            id: 'RXZlbnROb2RlOjE=',
            duration: 12,
            name: 'pentti',
            participantsPerInvite: EventParticipantsPerInvite.ChildAndGuardian,
            shortDescription: 'eventti',
            image:
              'http://localhost:8081/media/2020-02-15-184035_1920x1080_scrot.png',
            imageAltText: 'uooo',
          },
        },
      },
    },
  ],
};

const pastEvents: PastEvents = {
  edges: [
    {
      node: {
        id: 'RXZlbnROb2RlOjE=',
        participantsPerInvite: EventParticipantsPerInvite.Family,
        name: 'pentti',
        shortDescription: 'eventti',
        image:
          'http://localhost:8081/media/2020-02-15-184035_1920x1080_scrot.png',
        imageAltText: 'uooo',
        occurrences: {
          edges: [],
        },
      },
    },
  ],
};

const childWithEvents: ChildByIdResponse = {
  ...childData,
  upcomingEventsAndEventGroups,
  activeInternalAndTicketSystemEnrolments: enrolments,
  pastEvents: pastEvents,
};

const childOnlyAvailableEvents = {
  ...childData,
  upcomingEventsAndEventGroups,
  enrolments: {
    edges: [],
  },
  pastEvents: null,
};

const childOnlyEnrolments: ChildByIdResponse = {
  ...childData,
  upcomingEventsAndEventGroups: null,
  activeInternalAndTicketSystemEnrolments: {
    edges: [
      {
        node: {
          __typename: 'EnrolmentNode',
          id: 'foo',
          referenceId: 'bar',
          occurrence: {
            id: 'uu',
            time: '2020-02-24T09:09:09+00:00',
            venue: {
              id: '',
              address: '',
              name: '',
            },
            event: {
              id: 'RXZlbnROb2RlOjE=',
              duration: 12,
              name: 'pentti',
              participantsPerInvite:
                EventParticipantsPerInvite.ChildAndGuardian,
              shortDescription: 'eventti',
              image:
                'http://localhost:8081/media/2020-02-15-184035_1920x1080_scrot.png',
              imageAltText: 'uooo',
            },
          },
        },
      },
    ],
  },
  pastEvents: null,
};

const childOnlyPastEvents = {
  ...childData,
  upcomingEventsAndEventGroups: null,
  enrolments: {
    edges: [],
  },
  pastEvents: pastEvents,
};

test('Renders "No events" when no events"', () => {
  const { container } = render(<ProfileEvents child={childNoEvents} />);
  expect(container).toMatchSnapshot();
});

test('Renders events list when events of any type', () => {
  const { container } = render(<ProfileEvents child={childWithEvents} />);
  expect(container).toMatchSnapshot();
});

test('Renders events list when only upcomingEventsAndEventGroups', () => {
  const { container } = render(
    <ProfileEvents child={childOnlyAvailableEvents} />
  );
  expect(container).toMatchSnapshot();
});

test('Renders events list when only enrolments', () => {
  const { container } = render(<ProfileEvents child={childOnlyEnrolments} />);
  expect(container).toMatchSnapshot();
});

test('Renders events list when only past events', () => {
  const { container } = render(<ProfileEvents child={childOnlyPastEvents} />);
  expect(container).toMatchSnapshot();
});
