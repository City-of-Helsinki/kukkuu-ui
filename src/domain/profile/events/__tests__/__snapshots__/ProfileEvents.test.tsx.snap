// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders "No events" when no events" 1`] = `
<div>
  <div
    class="noEventWrapper"
  >
    <h2>
      Tapahtumakutsut
    </h2>
    <div
      class="noEvent"
    >
      <div
        class="imgWrapper envelopIcon"
      >
        <img
          alt=""
          src="envelopeWithStars.svg"
        />
      </div>
      <p>
        Tänne saapuvat sinulle lähetetyt tapahtumakutsut. Pääset myös ilmoittautumaan tapahtumiin ja perumaan ilmoittautumisen, jos et jostain syystä pääsekään osallistumaan tapahtumaan.
      </p>
    </div>
  </div>
</div>
`;

exports[`Renders events list when events of any type 1`] = `
<div>
  <ul
    class="list spacing-xl"
  >
    <li
      class="item"
    >
      <h2
        class="text h2 "
      >
        Tulevat tapahtumasi
      </h2>
      <ul
        class="list spacing-layout-2-xs"
      >
        <li
          class="item"
        >
          <div
            class="wrapper"
          >
            <div
              class="image"
            >
              <div
                class="qrWrapper"
              />
            </div>
            <div
              class="content"
            >
              <h3
                class="text h3 title"
              >
                pentti
              </h3>
              <p
                class="text body "
              >
                eventti
              </p>
              <div
                class="focalPoint"
              >
                <span>
                  <div
                    class="row"
                  >
                    <div
                      class="label fullWidth"
                      id="venue"
                    >
                      <svg
                        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fill-rule="evenodd"
                        >
                          <path
                            d="M0 0h24v24H0z"
                          />
                          <path
                            d="M11.967 1.5c2.06 0 4.12.778 5.69 2.334 3.143 3.111 2.93 7.96 0 11.268l-.622.709c-2.612 2.991-4.066 4.96-5.068 6.937-1.073-2.13-2.682-4.249-5.689-7.646-2.93-3.308-3.143-8.157 0-11.268A8.06 8.06 0 0111.967 1.5zm.032 2a6.072 6.072 0 00-4.3 1.762A5.606 5.606 0 006.002 9.41c.02 1.573.648 3.134 1.766 4.398l.66.752c1.59 1.823 2.717 3.239 3.573 4.503.975-1.437 2.292-3.063 4.233-5.255 1.118-1.264 1.746-2.825 1.766-4.398a5.616 5.616 0 00-1.698-4.15A6.077 6.077 0 0011.999 3.5zM12 6a3.5 3.5 0 110 6.999A3.5 3.5 0 0112 6zm0 2c-.827 0-1.5.673-1.5 1.5S11.173 11 12 11s1.5-.673 1.5-1.5S12.827 8 12 8z"
                            fill="currentColor"
                          />
                        </g>
                      </svg>
                      <div>
                        <p
                          class="text body label"
                        />
                      </div>
                    </div>
                    <div
                      class="label"
                      id="time"
                    >
                      <svg
                        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fill-rule="evenodd"
                        >
                          <path
                            d="M0 0h24v24H0z"
                          />
                          <path
                            d="M17 2a1 1 0 011 1v1h4v17H2V4h4V3a1 1 0 112 0v1h8V3a1 1 0 011-1zm3 9H4v8h16v-8zm0-5H4v3h16V6z"
                            fill="currentColor"
                          />
                        </g>
                      </svg>
                      <div>
                        <p
                          class="text body label"
                        >
                          24.2.2020
                        </p>
                      </div>
                    </div>
                    <div
                      class="label"
                      id="duration"
                    >
                      <svg
                        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fill-rule="evenodd"
                        >
                          <path
                            d="M0 0h24v24H0z"
                          />
                          <path
                            d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm1 2v5.584l3.243 3.244-1.415 1.415-3.535-3.536.002-.001-.294-.292L11 6h2z"
                            fill="currentColor"
                          />
                        </g>
                      </svg>
                      <div>
                        <p
                          class="text body label"
                        >
                          07:07 - 07:19
                        </p>
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </div>
            <div
              class="cta"
            >
              <button
                aria-label="Näytä tapahtuman tiedot"
                class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO actionWrapper"
                style="--color: var(--color-black); --color-hover: var(--color-black); --color-focus: var(--color-black-90); --color-hover-focus: var(--color-black-90); --background-color: transparent; --background-color-hover: transparent; --background-color-focus: transparent; --background-color-hover-focus: transparent; --background-selected: var(--color-summer); --border-color: none; --border-color-hover: var(--color-summer-dark); --border-color-focus: var(--color-summer-dark); --border-color-hover-focus: var(--color-summer-dark); --border-color-selected: var(--color-summer); --border-color-selected-hover: var(--color-summer-dark); --border-color-selected-focus: var(--color-summer); --border-color-selected-hover-focus: var(--color-summer);"
                type="button"
              >
                <span
                  class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                >
                  <svg
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <polygon
                        points="0 24 0 0 24 0 24 24"
                      />
                      <polygon
                        fill="currentColor"
                        points="13.5 12 8.5 7 10 5.5 16.5 12 10 18.5 8.5 17"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </li>
      </ul>
    </li>
    <li
      class="item"
    >
      <h2
        class="text h2 "
      >
        Tapahtumakutsut
      </h2>
      <ul
        class="list spacing-layout-2-xs"
      >
        <li
          class="item"
        >
          <div
            class="wrapper"
          >
            <div
              class="image"
            >
              <img
                alt="uooo"
                class="image"
                src="http://localhost:8081/media/2020-02-15-184035_1920x1080_scrot.png"
              />
            </div>
            <div
              class="content"
            >
              <h3
                class="text h3 title"
              >
                pentti
              </h3>
              <p
                class="text body "
              >
                eventti
              </p>
              <div
                class="focalPoint"
              >
                <button
                  class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO primaryActionButton"
                  style="--color: var(--color-black); --color-hover: var(--color-black); --color-focus: var(--color-black-90); --color-hover-focus: var(--color-black-90); --background-color: var(--color-summer); --background-color-hover: var(--color-summer-dark); --background-color-focus: var(--color-summer); --background-color-hover-focus: var(--color-summer-dark); --background-selected: var(--color-summer); --border-color: var(--color-summer); --border-color-hover: var(--color-summer-dark); --border-color-focus: var(--color-summer-dark); --border-color-hover-focus: var(--color-summer-dark); --border-color-selected: var(--color-summer); --border-color-selected-hover: var(--color-summer-dark); --border-color-selected-focus: var(--color-summer); --border-color-selected-hover-focus: var(--color-summer);"
                  type="button"
                >
                  <span
                    class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                  >
                    Lue lisää
                  </span>
                </button>
                <span />
              </div>
            </div>
            <div
              class="cta"
            >
              <button
                aria-label="Lue lisää"
                class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO actionWrapper"
                style="--color: var(--color-black); --color-hover: var(--color-black); --color-focus: var(--color-black-90); --color-hover-focus: var(--color-black-90); --background-color: transparent; --background-color-hover: transparent; --background-color-focus: transparent; --background-color-hover-focus: transparent; --background-selected: var(--color-summer); --border-color: none; --border-color-hover: var(--color-summer-dark); --border-color-focus: var(--color-summer-dark); --border-color-hover-focus: var(--color-summer-dark); --border-color-selected: var(--color-summer); --border-color-selected-hover: var(--color-summer-dark); --border-color-selected-focus: var(--color-summer); --border-color-selected-hover-focus: var(--color-summer);"
                type="button"
              >
                <span
                  class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                >
                  <svg
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <polygon
                        points="0 24 0 0 24 0 24 24"
                      />
                      <polygon
                        fill="currentColor"
                        points="13.5 12 8.5 7 10 5.5 16.5 12 10 18.5 8.5 17"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </li>
      </ul>
    </li>
    <li
      class="item"
    >
      <h2
        class="text h2 "
      >
        Menneet tapahtumat
      </h2>
      <ul
        class="list spacing-layout-2-xs"
      >
        <li
          class="item"
        >
          <div
            class="wrapper"
          >
            <div
              class="image"
            >
              <img
                alt="uooo"
                class="image"
                src="http://localhost:8081/media/2020-02-15-184035_1920x1080_scrot.png"
              />
            </div>
            <div
              class="content"
            >
              <h3
                class="text h3 title"
              >
                pentti
              </h3>
              <p
                class="text body "
              >
                eventti
              </p>
              <div
                class="focalPoint"
              >
                <span />
              </div>
            </div>
            <div
              class="cta"
            >
              <button
                aria-label="Näytä tapahtuman tiedot"
                class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO actionWrapper"
                style="--color: var(--color-black); --color-hover: var(--color-black); --color-focus: var(--color-black-90); --color-hover-focus: var(--color-black-90); --background-color: transparent; --background-color-hover: transparent; --background-color-focus: transparent; --background-color-hover-focus: transparent; --background-selected: var(--color-summer); --border-color: none; --border-color-hover: var(--color-summer-dark); --border-color-focus: var(--color-summer-dark); --border-color-hover-focus: var(--color-summer-dark); --border-color-selected: var(--color-summer); --border-color-selected-hover: var(--color-summer-dark); --border-color-selected-focus: var(--color-summer); --border-color-selected-hover-focus: var(--color-summer);"
                type="button"
              >
                <span
                  class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                >
                  <svg
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <polygon
                        points="0 24 0 0 24 0 24 24"
                      />
                      <polygon
                        fill="currentColor"
                        points="13.5 12 8.5 7 10 5.5 16.5 12 10 18.5 8.5 17"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`Renders events list when only enrolments 1`] = `
<div>
  <ul
    class="list spacing-xl"
  >
    <li
      class="item"
    >
      <h2
        class="text h2 "
      >
        Tulevat tapahtumasi
      </h2>
      <ul
        class="list spacing-layout-2-xs"
      >
        <li
          class="item"
        >
          <div
            class="wrapper"
          >
            <div
              class="image"
            >
              <div
                class="qrWrapper"
              />
            </div>
            <div
              class="content"
            >
              <h3
                class="text h3 title"
              >
                pentti
              </h3>
              <p
                class="text body "
              >
                eventti
              </p>
              <div
                class="focalPoint"
              >
                <span>
                  <div
                    class="row"
                  >
                    <div
                      class="label fullWidth"
                      id="venue"
                    >
                      <svg
                        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fill-rule="evenodd"
                        >
                          <path
                            d="M0 0h24v24H0z"
                          />
                          <path
                            d="M11.967 1.5c2.06 0 4.12.778 5.69 2.334 3.143 3.111 2.93 7.96 0 11.268l-.622.709c-2.612 2.991-4.066 4.96-5.068 6.937-1.073-2.13-2.682-4.249-5.689-7.646-2.93-3.308-3.143-8.157 0-11.268A8.06 8.06 0 0111.967 1.5zm.032 2a6.072 6.072 0 00-4.3 1.762A5.606 5.606 0 006.002 9.41c.02 1.573.648 3.134 1.766 4.398l.66.752c1.59 1.823 2.717 3.239 3.573 4.503.975-1.437 2.292-3.063 4.233-5.255 1.118-1.264 1.746-2.825 1.766-4.398a5.616 5.616 0 00-1.698-4.15A6.077 6.077 0 0011.999 3.5zM12 6a3.5 3.5 0 110 6.999A3.5 3.5 0 0112 6zm0 2c-.827 0-1.5.673-1.5 1.5S11.173 11 12 11s1.5-.673 1.5-1.5S12.827 8 12 8z"
                            fill="currentColor"
                          />
                        </g>
                      </svg>
                      <div>
                        <p
                          class="text body label"
                        />
                      </div>
                    </div>
                    <div
                      class="label"
                      id="time"
                    >
                      <svg
                        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fill-rule="evenodd"
                        >
                          <path
                            d="M0 0h24v24H0z"
                          />
                          <path
                            d="M17 2a1 1 0 011 1v1h4v17H2V4h4V3a1 1 0 112 0v1h8V3a1 1 0 011-1zm3 9H4v8h16v-8zm0-5H4v3h16V6z"
                            fill="currentColor"
                          />
                        </g>
                      </svg>
                      <div>
                        <p
                          class="text body label"
                        >
                          24.2.2020
                        </p>
                      </div>
                    </div>
                    <div
                      class="label"
                      id="duration"
                    >
                      <svg
                        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fill-rule="evenodd"
                        >
                          <path
                            d="M0 0h24v24H0z"
                          />
                          <path
                            d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm1 2v5.584l3.243 3.244-1.415 1.415-3.535-3.536.002-.001-.294-.292L11 6h2z"
                            fill="currentColor"
                          />
                        </g>
                      </svg>
                      <div>
                        <p
                          class="text body label"
                        >
                          09:09 - 09:21
                        </p>
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </div>
            <div
              class="cta"
            >
              <button
                aria-label="Näytä tapahtuman tiedot"
                class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO actionWrapper"
                style="--color: var(--color-black); --color-hover: var(--color-black); --color-focus: var(--color-black-90); --color-hover-focus: var(--color-black-90); --background-color: transparent; --background-color-hover: transparent; --background-color-focus: transparent; --background-color-hover-focus: transparent; --background-selected: var(--color-summer); --border-color: none; --border-color-hover: var(--color-summer-dark); --border-color-focus: var(--color-summer-dark); --border-color-hover-focus: var(--color-summer-dark); --border-color-selected: var(--color-summer); --border-color-selected-hover: var(--color-summer-dark); --border-color-selected-focus: var(--color-summer); --border-color-selected-hover-focus: var(--color-summer);"
                type="button"
              >
                <span
                  class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                >
                  <svg
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <polygon
                        points="0 24 0 0 24 0 24 24"
                      />
                      <polygon
                        fill="currentColor"
                        points="13.5 12 8.5 7 10 5.5 16.5 12 10 18.5 8.5 17"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`Renders events list when only past events 1`] = `
<div>
  <ul
    class="list spacing-xl"
  >
    <li
      class="item"
    >
      <h2
        class="text h2 "
      >
        Menneet tapahtumat
      </h2>
      <ul
        class="list spacing-layout-2-xs"
      >
        <li
          class="item"
        >
          <div
            class="wrapper"
          >
            <div
              class="image"
            >
              <img
                alt="uooo"
                class="image"
                src="http://localhost:8081/media/2020-02-15-184035_1920x1080_scrot.png"
              />
            </div>
            <div
              class="content"
            >
              <h3
                class="text h3 title"
              >
                pentti
              </h3>
              <p
                class="text body "
              >
                eventti
              </p>
              <div
                class="focalPoint"
              >
                <span />
              </div>
            </div>
            <div
              class="cta"
            >
              <button
                aria-label="Näytä tapahtuman tiedot"
                class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO actionWrapper"
                style="--color: var(--color-black); --color-hover: var(--color-black); --color-focus: var(--color-black-90); --color-hover-focus: var(--color-black-90); --background-color: transparent; --background-color-hover: transparent; --background-color-focus: transparent; --background-color-hover-focus: transparent; --background-selected: var(--color-summer); --border-color: none; --border-color-hover: var(--color-summer-dark); --border-color-focus: var(--color-summer-dark); --border-color-hover-focus: var(--color-summer-dark); --border-color-selected: var(--color-summer); --border-color-selected-hover: var(--color-summer-dark); --border-color-selected-focus: var(--color-summer); --border-color-selected-hover-focus: var(--color-summer);"
                type="button"
              >
                <span
                  class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                >
                  <svg
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <polygon
                        points="0 24 0 0 24 0 24 24"
                      />
                      <polygon
                        fill="currentColor"
                        points="13.5 12 8.5 7 10 5.5 16.5 12 10 18.5 8.5 17"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`Renders events list when only upcomingEventsAndEventGroups 1`] = `
<div>
  <ul
    class="list spacing-xl"
  >
    <li
      class="item"
    >
      <h2
        class="text h2 "
      >
        Tapahtumakutsut
      </h2>
      <ul
        class="list spacing-layout-2-xs"
      >
        <li
          class="item"
        >
          <div
            class="wrapper"
          >
            <div
              class="image"
            >
              <img
                alt="uooo"
                class="image"
                src="http://localhost:8081/media/2020-02-15-184035_1920x1080_scrot.png"
              />
            </div>
            <div
              class="content"
            >
              <h3
                class="text h3 title"
              >
                pentti
              </h3>
              <p
                class="text body "
              >
                eventti
              </p>
              <div
                class="focalPoint"
              >
                <button
                  class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO primaryActionButton"
                  style="--color: var(--color-black); --color-hover: var(--color-black); --color-focus: var(--color-black-90); --color-hover-focus: var(--color-black-90); --background-color: var(--color-summer); --background-color-hover: var(--color-summer-dark); --background-color-focus: var(--color-summer); --background-color-hover-focus: var(--color-summer-dark); --background-selected: var(--color-summer); --border-color: var(--color-summer); --border-color-hover: var(--color-summer-dark); --border-color-focus: var(--color-summer-dark); --border-color-hover-focus: var(--color-summer-dark); --border-color-selected: var(--color-summer); --border-color-selected-hover: var(--color-summer-dark); --border-color-selected-focus: var(--color-summer); --border-color-selected-hover-focus: var(--color-summer);"
                  type="button"
                >
                  <span
                    class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                  >
                    Lue lisää
                  </span>
                </button>
                <span />
              </div>
            </div>
            <div
              class="cta"
            >
              <button
                aria-label="Lue lisää"
                class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO actionWrapper"
                style="--color: var(--color-black); --color-hover: var(--color-black); --color-focus: var(--color-black-90); --color-hover-focus: var(--color-black-90); --background-color: transparent; --background-color-hover: transparent; --background-color-focus: transparent; --background-color-hover-focus: transparent; --background-selected: var(--color-summer); --border-color: none; --border-color-hover: var(--color-summer-dark); --border-color-focus: var(--color-summer-dark); --border-color-hover-focus: var(--color-summer-dark); --border-color-selected: var(--color-summer); --border-color-selected-hover: var(--color-summer-dark); --border-color-selected-focus: var(--color-summer); --border-color-selected-hover-focus: var(--color-summer);"
                type="button"
              >
                <span
                  class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                >
                  <svg
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <polygon
                        points="0 24 0 0 24 0 24 24"
                      />
                      <polygon
                        fill="currentColor"
                        points="13.5 12 8.5 7 10 5.5 16.5 12 10 18.5 8.5 17"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
